tensordict/utils.py

```diff
@@ -1851,6 +1851,85 @@ def format_size(size):
         logging.info(indent + os.path.basename(path))
 
 
+def isin(
+    tensordict: TensorDictBase,
+    reference_tensordict: TensorDictBase,
+    key: str,
+    dim: int = 0,
+    invert: bool = False,
+):
+    """Tests if each element of 'key' in tensordict 'dim' is also present in the reference_tensordict.
+
+    This function returns a boolean tensor of length  ´tensordict.batch_size[dim]´ that is True for elements in
+    'key' that are also present in the reference_tensordict. This function assumes that both tensordict and
+    reference_tensordict have the same batch size and contain the specified key, otherwise it will raise an error.
+
+    Args:
+        tensordict (TensorDictBase): Input TensorDict.
+        reference_tensordict (TensorDictBase): Target TensorDict against which to test.
+        key (str): The key to test.
+        dim (int): The dimension along which to test. Default: 0
+        invert (bool):  If True, inverts the boolean return tensor, resulting in True values for elements
+            not in reference_tensordict. Default: False
+
+    Returns:
+        out (Tensor): A boolean tensor of length ´tensordict.batch_size[dim]´ that is True for elements in
+            'key' that are also present in the reference_tensordict or False if invert is True.
+
+    Examples:
+        >>> td = TensorDict(
+        ...     {
+        ...         "tensor1": torch.tensor([[1, 2, 3], [4, 5, 6], [1, 2, 3], [7, 8, 9]]),
+        ...         "tensor2": torch.tensor([[10, 20], [30, 40], [40, 50], [50, 60]]),
+        ...     },
+        ...     batch_size=[4],
+        ... )
+        >>> td_ref = TensorDict(
+        ...     {
+        ...         "tensor1": torch.tensor([[1, 2, 3], [4, 5, 6], [10, 11, 12]]),
+        ...         "tensor2": torch.tensor([[10, 20], [30, 40], [50, 60]]),
+        ...     },
+        ...     batch_size=[3],
+        ... )
+        >>> in_reference = is_in_reference(td, td_ref, key="tensor1")
+        >>> expected_in_reference = torch.tensor([True, True, True, False])
+        >>> assert torch.equal(in_reference, expected_in_reference)
+    """
+    # Check key is present in both tensordict and reference_tensordict
+    if key not in tensordict.keys(include_nested=True):
+        raise KeyError(f"Key '{key}' not found in tensordict.")
+    if key not in reference_tensordict.keys(include_nested=True):
+        raise KeyError(f"Key '{key}' not found in reference_tensordict.")
```

Comment:
What I usually do (to check if the key is there only once and make things faster)
```suggestion
    reference_tensor = reference_tensordict.get(key, default=None)
    target_tensor = tensordict.get(key, default=None)

    if not isinstance(target_tensor, torch.Tensor):
        raise KeyError(f"Key '{key}' not found in tensordict or not a tensor.")
    if not isinstance(reference_tensor, torch.Tensor):
        raise KeyError(f"Key '{key}' not found in reference_tensordict or not a tensor.")
```